pandas/io/excel/_base.py

```diff
@@ -896,6 +907,17 @@ def __init__(
 
         self.mode = mode
 
+        if if_sheet_exists not in [None, "error", "new", "replace"]:
+            raise ValueError(
+                f"'{if_sheet_exists}' is not valid for if_sheet_exists. "
+                "Valid options are 'error', 'new' and 'replace'."
+            )
+        if if_sheet_exists and "r+" not in mode:
+            raise ValueError("if_sheet_exists is only valid in append mode (mode='a')")
+        if if_sheet_exists is None and "r+" in mode:
```

Comment:
why is the `r+` condition here? (e.g. then if_sheet_exists) should always be a str at this point, right? (if its being written will be ignored anyhow), right?

Reply:
i guess not a big deal

Reply:
You're right, fixed